lib/logstorage/bloomfilter.go

```diff
@@ -82,12 +83,33 @@ func (bf *bloomFilter) mustInitTokens(tokens []string) {
 // mustInitHashes initializes bf with the given hashes
 func (bf *bloomFilter) mustInitHashes(hashes []uint64) {
 	bitsCount := len(hashes) * bloomFilterBitsPerItem
-	wordsCount := (bitsCount + 63) / 64
-	bits := slicesutil.SetLength(bf.bits, wordsCount)
+	wordsCount := int64((bitsCount + 63) / 64)
+	/*
+		If the length of the array is a power of 2, then subsequent calculations can use bitwise operations instead of modulo operations to improve performance.
+		However, the increase in array length also increases the length of the data, resulting in negative optimization.
+		A wider range of trade-offs may be required.
+	*/
+	//wordsCount = roundUpPowerOf2(wordsCount)
+	bits := slicesutil.SetLength(bf.bits, int(wordsCount))
 	bloomFilterAddHashes(bits, hashes)
 	bf.bits = bits
 }
 
+func roundUpPowerOf2(x int64) int64 {
```

Comment:
This function isn't used, so it must be removed. Also, it might be better from the readability and performance PoV to use [`bits.Len64()`](https://pkg.go.dev/math/bits#Len64) instead of the manual calculations (if this function will be used in the future):

```go
func roundUpPowerOf(x int64) int64 {
  return int64(1 << bits.Len(uint64(x)))
}
```